docs/DeveloperGuide.adoc

```diff
@@ -0,0 +1,665 @@
+
+=== Prerequisites
+. JDK 11 or above
+. IntelliJ IDE
+
+=== Setting up the project
+. Fork this repository, and clone the fork to your computer
+. Open the IntelliJ IDE. If you are not in the welcome screen, click `File` > `Close Project` to close the existing project.
+. Set up the correct JDK version for Gradle
+.. Click `Configure` > `Project Defaults` > `Project Structure`
+.. Click `New...` and find the directory of the JDK
+. Click on `Import Project`
+. Locate and select the `build.gradle` file, then click `OK`
+. Click `Open as Project`
+. Click `OK` to use the default settings provided
+
+=== Verifying the Setup
+. In an IntelliJ terminal, run gradlew build
+. Cd to the folder `build` > `libs` and run: `java -jar atas-2.0.0.jar`
+.. To use *ATAS*, simply type a valid command into the terminal and press kbd:[Enter] to run the command. +
+e.g. Typing `help` command and pressing kbd:[Enter] will list the commands present
+.. Some example commands you can try to get familiar with *ATAS*:
+* `help`: Lists the commands that *ATAS* supports.
+* `assignment n/Assignment One m/CS2113T d/01/01/20 1600 c/Important Assignment`: Adds an assignment called *Assignment
+one* for the module *CS2113T*. This assignment is due on *01/01/2020 1600* and the comments for this assignment is that
+it is an *Important Assignment*.
+* `exit`: Exits *ATAS*.
+
+A summary of all the features available in *ATAS* can be found in <<Commands Summary>>. +
+Refer to <<Features>> for the detailed instruction of the various commands of *ATAS*.
+
+== Design
+This section will give a high-level overview of how various components in *ATAS* function and interact with each other.
+
+=== Architecture
+image::overall architecture.PNG[overall architecture]
+The architecture diagram above illustrates the high-level design of the *ATAS* application. +
+
+The `Atas` component contains all the other components in the application. +
+
+* `Ui`: Reads user input, and shows the results of commands to the user
+* `Logic`: Handles execution of user input commands
+* `Model`: Stores the runtime data of the application
+* `Storage`: Reads and stores data from a file stored on the user's computer
+
+The following sections will explain each component in greater detail.
+
+=== UI Component
+1. The `Ui` component reads user input which represents a command.
+2. The `Ui` component shows the result of the command to the user.
+
+=== Logic Component
+The Logic component comprises the `Parser`, `Command`, and `CommandResult` classes: +
+
+1. `Parser` will interpret the user command and return the corresponding `Command` object. +
+2. `Command#execute()` is called to run the command, returning a `CommandResult` object. +
+3. The `CommandResult` object will contain the output to be shown to the user.
+
+=== Model Component
+The Model component contains the `Task` and `TaskList` classes, which store the user's schedule.
+
+image::TaskList Task class diagram.PNG[TaskList and Tasks]
```

Comment:
This diagram gives a very good understanding of your program! Also, all the arrows look correct to me!